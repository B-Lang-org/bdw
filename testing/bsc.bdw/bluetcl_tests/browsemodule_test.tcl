namespace import ::Bluetcl::*

# Sets the specified flag
puts [flags set {-verilog}]

# Loads the specified module
puts [module load mkT]

# Shows the module hierarchy
puts [browsemodule list 0]

# Expands the specified node of the module hierarchy
puts [browsemodule list 1]
puts [browsemodule list 2]
puts [browsemodule list 5]

# Returns information concerning the specified node of the module hierarchy
puts [browsemodule detail 5]
puts [browsemodule detail 2]
puts [browsemodule detail 1]

# Loads the specified module
puts [module load mkM]

# Refreshes the module hierarchy(used when the loaded module has changed)
puts [browsemodule refresh]

# Shows the module hierarchy
puts [browsemodule list 0]

