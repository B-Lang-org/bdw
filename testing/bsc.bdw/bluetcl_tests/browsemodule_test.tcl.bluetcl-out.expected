-verilog
mkM mkT
{1 {top (mkT)} {}}
{2 {r1 (mkM)} {}} {3 {r2 (mkM)} {}}
{4 {ptr (RegN)} {}} {5 {r1 (RegN)} {}} {6 {r2 (RegN)} {}}

primitive
user {instance r1} {module mkM} {interface Reg#(Bool)} {methods {_write _read}} {rules {RL_rHello RL_rWorld}} {schedule {parallel: [esposito: [_write -> [], _read -> [], RL_rHello -> [], RL_rWorld -> []]]
order: [_read, RL_rHello, _write, RL_rWorld]}}
user {instance top} {module mkT} {interface Reg#(Bool)} {methods {_write _read}} {rules {}} {schedule {parallel: [esposito: [_write -> [], _read -> []]]
order: [_write, _read]}}
mkM

{1 {top (mkM)} {}}
